### PR TITLE
Add support for Customer Balance Transaction resource and APIs

### PR DESCRIPTION
**Pull request by remi-stripe**
r? @ob-stripe 
cc @stripe/api-libraries

### CODE REVIEW

**Comment by remi-stripe** [comment]
@ob-stripe PTAL

File: test/resources/CustomerBalanceTransactions.spec.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+var resources = require('../../lib/stripe').resources;
+var stripe = require('../../testUtils').getSpyableStripe();
+var expect = require('chai').expect;
+
+var CUSTOMER_TEST_ID = 'cus_123';
+var TRANSACTION_TEST_ID = 'cbtxn_123';
+
+var taxId = new resources.CustomerBalanceTransactions(
```
Comment (ob-stripe) [inline test/resources/CustomerBalanceTransactions.spec.js]: Can you rename `taxId` to `customerBalanceTransaction` or `transaction`?

File: lib/resources/CustomerBalanceTransactions.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+/**
+ * CustomerBalanceTransactions is a unique resource in that, upon instantiation,
+ * requires a customerId, and therefore each of its methods only
+ * require the transactionId argument.
+ *
+ * This streamlines the API specifically for the case of accessing a customer balance tranaction
+ * on a returned customer object.
+ *
+ * E.g. customerObject.transactions.retrieve(transactionId)
+ * (As opposed to the also-supported stripe.customers.retrieveTaxId(customerId, transactionId))
```
Comment (ob-stripe) [inline lib/resources/CustomerBalanceTransactions.js]:
```suggestion
 * (As opposed to the also-supported stripe.customers.retrieveBalanceTransaction(customerId, transactionId))
```

**Comment by remi-stripe** [comment]
PTAL @ob-stripe since I rebased on latest master and re-did the PR

File: lib/resources/Customers.js [hunk 3]
```diff
@@ -64,4 +64,28 @@ module.exports = StripeResource.extend({
     method: 'GET',
     path: '/{customer}/tax_ids/{id}',
   }),
+
+  createBalanceTransaction: stripeMethod({
+    method: 'POST',
+    path: '/{customer}/balance_transactions',
+    urlParams: ['customer'],
```
Comment (ob-stripe) [inline lib/resources/Customers.js]: `urlParams` is no longer needed, can you remove it from all methods?

**Comment by remi-stripe** [comment]
@ob-stripe fixed, PTAL

**Comment by ob-stripe** [comment]
Released as 7.2.0.

**Pull request merged**